### ISSUE TITLE
Appsody deploy must allow specifying push and pull image registry URLs

### ISSUE THREAD

**Issue by chilanti** [appsody/appsody#451]
**Is your feature request related to a problem? Please describe.**
This issue is related to #410: when the image registry is hosted on a private cloud or on a shared K8S cluster, it may be accessible through different URLs: 
* an externally addressable URL, for developers who need to push their image to the registry
* an internally addressable URL, which is visibile only within the K8S cluster, for pods that need to pull the image at container creation time.

Currently `appsody deploy` doesn't support specifying these two URLs, causing either a docker push failure as the command is run, or a docker pull failure when the contained in being created after deployment.

**Describe the solution you'd like**
Provide the ability to specify a "push registry URL" and a "pull registry URL" in `appsody deploy`.  The push URL is used to executed the `docker push` command. The pull URL is stored in the `app-deploy.yaml` manifest.
To recap - users will be able to run: 
* `appsody deploy -t <my-repo>/<my-image> --push`
just as they do today. No changes here - the image will be pushed to the default registry and it is assumed to be pulled from that same registry by the K8S cluster (this is the docker.io use case).
* `appsody deploy -t <my-repo>/<my-image> --push-url <external registry URL> --pull-url <internal registry URL>`
which causes the image to be pushed to `<external registry URL>/<my-repo>/<my-image>`, and to be pulled from `<internal registry URL>/<my-repo>/<my-image>`
* `appsody/deploy` will still call the image `dev.local/<image name>` - no changes.

**Describe alternatives you've considered**

**Additional context**

**Comment by csantanapr** [comment]
This would be great to specify the image url where to push like today `--tag` and a new flag like `--image` on which image to put in the appsodyapplication cr

For now if people are hitting this problem, this is the only workaround I was able to make here https://ibm-cloud-architecture.github.io/Learning-Kabanero-101/web/1.0.0/appsody-deploy.html#_automate_appsody_deploy

**Comment by stevemar** [comment]
+1 to addressing this feature. I just ran into it today. @csantanapr's workaround was very helpful, thanks Carlos!
